internal/provider/resources/install/common.go

```diff
@@ -52,7 +59,33 @@ func (i *Install) reportConversionError(header string, subheader string, value a
 }
 
 func (i *Install) itemPath(id string) string {
-	return fmt.Sprintf("integrations/%s/config/%s/%s", i.Integration, i.Component, id)
+	return fmt.Sprintf("%s/%s/%s", i.itemBasePath(), i.Component, id)
+}
+
+func (i *Install) itemBasePath() string {
+	return fmt.Sprintf("integrations/%s/config", i.Integration)
+}
+
+// Places the item in the "stage" state in P0.
+// To use, the item's TFSDK model must be passed. For example:
+//
+//	var data ItemConfigurationModel
+//	var json ConfigurationApiResponseJson
+func (i *Install) Upsert(ctx context.Context, diags *diag.Diagnostics, plan *tfsdk.Plan, state *tfsdk.State, json any, model any) {
+	diags.Append(plan.Get(ctx, model)...)
+	if diags.HasError() {
+		return
+	}
+
+	throwaway_response := struct{}{}
+	err := i.ProviderData.Post(i.itemBasePath(), struct{}{}, &throwaway_response)
+	if err != nil {
+		// we can safely 409 Conflict errors, because they indicate the item is already installed
```

Comment:
I don't think this is true? It's a PUT, it shouldn't return 409s as it is idempotent. A 409 would be for some other reason than the item is already installed.